### ISSUE TITLE
ComboBox Reactivity Issues

### ISSUE THREAD

**Issue by DaleCam** [carbon-design-system/carbon-components-svelte#1262]
Hi there, first of all, all your work in this project is very appreciated! 
I seem to have found a fairly wierd issue with Combo boxes. If they are moved within the DOM a couple of times they lose their selectedId bindings!
Here is a REPL to demonstrate: https://stackblitz.com/edit/sveltejs-kit-template-default-6poqii?file=src/routes/index.svelte
If you hit the "Reorder" button a few times, you will notice the comboBox will lose its selected Id at some point.
However changing the combobox to a carbon component Select component (to ensure the issue is only for Comboboxes), the bound value persists!. So there is something wierd going on only in the ComboBox component... or the way svelte interacts with it...

**Comment by DaleCam** [comment]
If the stackblitz link doesnt start correctly (it has a few glitches) you can view the result here: https://sveltejs-kit-template-default-6poqii--3000.local-corp.webcontainer.io

**Comment by DaleCam** [comment]
Update: If you await tick() between each array update (ie between the 2 lines below) it solves the issue:
```   
   parentItems = [...parentItems.slice(0, from), ...parentItems.slice(from + 1)];
   await tick();
   parentItems = [...parentItems.slice(0, to), temp, ...parentItems.slice(to)];
```
which makes me feel there may be a need to await tick() somewhere in the combo box reactivity to avoid this issue. And i also coudn't figure out why carbon Select component doesn't have the same issue.

**Comment by DaleCam** [comment]
It appears it could be related to this svelte issue: https://github.com/sveltejs/svelte/issues/7578

**Comment by metonym** [comment]
Nice find. This can be fixed by using the native `bind:` directive instead of manually managing the `value` state.

**Comment by metonym** [comment]
Fixed in [v0.65.2](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.65.2).

@DaleCam Thanks for getting to the bottom of this, and for providing a thorough repro.
